src/core/agent/services/identifierService.ts

```diff
@@ -191,10 +195,22 @@ class IdentifierService extends AgentService {
     const metadata = await this.identifierStorage.getIdentifierMetadata(
       identifier
     );
-    this.validArchivedIdentifier(metadata);
-    await this.identifierStorage.updateIdentifierMetadata(identifier, {
-      isDeleted: true,
-    });
+
+    if (metadata.groupMetadata) {
```

Comment:
This only covers the 3rd AC of the story, not the first 2. You need to also check if `multisigManageAid` is set, and if it is do the same (AC 1), and also delete the manage AID.